spring-cloud-aws-parameter-store-config/src/test/java/org/springframework/cloud/aws/paramstore/AwsParamStorePropertySourceLocatorTest.java

```diff
@@ -0,0 +1,141 @@
+import com.amazonaws.services.simplesystemsmanagement.model.Parameter;
+import org.junit.Test;
+
+import org.springframework.mock.env.MockEnvironment;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+/**
+ * Unit test for {@link AwsParamStorePropertySourceLocator}.
+ *
+ * @author Matej Nedic
+ */
+public class AwsParamStorePropertySourceLocatorTest {
+
+	private AWSSimpleSystemsManagement ssmClient = mock(AWSSimpleSystemsManagement.class);
+
+	private MockEnvironment env = new MockEnvironment();
+
+	@Test
+	public void contextExpectedToHave2Elements() {
+		AwsParamStoreProperties properties = new AwsParamStorePropertiesBuilder()
+				.withDefaultContext("application").withName("application").build();
+
+		GetParametersByPathResult firstResult = getFirstResult();
+		GetParametersByPathResult nextResult = getNextResult();
+		when(ssmClient.getParametersByPath(any(GetParametersByPathRequest.class)))
+				.thenReturn(firstResult, nextResult);
+
+		AwsParamStorePropertySourceLocator locator = new AwsParamStorePropertySourceLocator(ssmClient, properties);
+		env.setActiveProfiles("test");
+		locator.locate(env);
+
+		assertThat(locator.getContexts().size()).isEqualTo(2);
+	}
+
+	@Test
+	public void contextExpectedToHave4Elements() {
+		AwsParamStoreProperties properties = new AwsParamStorePropertiesBuilder()
+				.withDefaultContext("application").withName("messaging-service").build();
+
+		GetParametersByPathResult firstResult = getFirstResult();
+		GetParametersByPathResult nextResult = getNextResult();
+		when(ssmClient.getParametersByPath(any(GetParametersByPathRequest.class)))
+				.thenReturn(firstResult, nextResult);
+
+		AwsParamStorePropertySourceLocator locator = new AwsParamStorePropertySourceLocator(
+				ssmClient, properties);
+		env.setActiveProfiles("test");
+		locator.locate(env);
+
+		assertThat(locator.getContexts().size()).isEqualTo(4);
+	}
+
+	@Test
+	public void contextSpecificOrderExpected() {
+		AwsParamStoreProperties properties = new AwsParamStorePropertiesBuilder()
+			.withDefaultContext("application")
+			.withName("messaging-service").build();
+
+		GetParametersByPathResult firstResult = getFirstResult();
+		GetParametersByPathResult nextResult = getNextResult();
+		when(ssmClient.getParametersByPath(any(GetParametersByPathRequest.class)))
+			.thenReturn(firstResult, nextResult);
+
+		AwsParamStorePropertySourceLocator locator = new AwsParamStorePropertySourceLocator(
+			ssmClient, properties);
+		env.setActiveProfiles("test");
+		locator.locate(env);
+
+
+		List contextToBeTested =  new ArrayList(locator.getContexts());
+
+		assertThat(contextToBeTested.get(0)).isEqualTo("application/messaging-service_test/");
+		assertThat(contextToBeTested.get(1)).isEqualTo("application/application_test/");
```

Comment:
If you run this test against current version without your changes, the order will be different:

1. application specific profiles
2. application
3. default profiles
4. default

So index `1` is flipped with `2`.

Reply:
I saw same behaviour as well, but since you wrote that profiles should go first I understood that both profiles should be followed by application name. Sorry for missunderstanding. Should I change the order?

Reply:
I saw same behaviour as well, but since you wrote that profiles should go first I understood that both profiles should be followed by application name. Sorry for missunderstanding. Should I change the order?